### PR TITLE
Level order iterator for optimization functions

### PR DESCRIPTION
**Pull request by invokesus**

### CODE REVIEW

**Comment by domoritz** [comment]
@invokesus What's the status of this PR?

**Comment by invokesus** [comment]
Will work on it this weekend.

**Comment by domoritz** [comment]
Sounds good. Ping me when you want me to look at it.

**Comment by kanitw** [comment]
Is this ready?

**Comment by kanitw** [comment]
(If so, rebase so we can review?)

**Comment by domoritz** [comment]
this PR is supposed to address https://github.com/vega/vega-lite/issues/4174

File: src/compile/data/optimizers.ts [hunk 1]
```diff
@@ -248,19 +247,40 @@ export class RemoveUnnecessaryNodes extends TopDownOptimizer {
...
     return this.mutatedFlag;
   }
 }
 
 /**
  * Inserts an Intermediate ParseNode containing all non-conflicting Parse fields and removes the empty ParseNodes
  */
-export class MergeParse extends BottomUpOptimizer {
-  public run(node: DataFlowNode): optimizers.OptimizerFlags {
+export class MergeParse extends LevelOrderOptimizer {
```
Comment (domoritz) [inline src/compile/data/optimizers.ts]: Ideally, the implementation here wouldn't need to change.

**Pull request closed without merging**